x/feedist/model.go

```diff
@@ -0,0 +1,95 @@
+package feedist
+
+import (
+	"fmt"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+)
+
+var _ orm.CloneableData = (*Revenue)(nil)
+
+func (rev *Revenue) Validate() error {
+	if err := rev.Admin.Validate(); err != nil {
+		return errors.Wrap(err, "invalid admin signature")
+	}
+	if len(rev.Recipients) == 0 {
+		return errors.InvalidModelErr.New("no recipients")
+	}
+	for i, r := range rev.Recipients {
+		if err := r.Address.Validate(); err != nil {
+			return errors.Wrap(err, fmt.Sprintf("recipient %d address", i))
```

Comment:
Off-topic:
This is an example of an error that is marked internal error and would have been replaced with the string "internal error" in the original errors pr. However, we DO want to send this information to the client. This was the cause of my feedback

Reply:
I believe that would not be an internal error, because [`Address.Validate` returns a weave error](https://github.com/iov-one/weave/blob/master/conditions.go#L112).

Below code would return an internal error
```go
if _, err := os.Open("/proc/1/stat"); err != nil {
    return errors.Wrap(err, "fail")
}
```
This could be a topic for another discussion if it will become a problem. :wink: